src/types/view-devices/vDevice.ts

```diff
@@ -400,6 +405,72 @@ export abstract class ViewDevice extends Container {
     RightBar.getInstance().renderInfo(new DeviceInfo(this));
   }
 
+  private repositionDeviceIcons() {
+    const icons = Object.values(this.deviceIcons).filter(Boolean) as Text[];
+    if (icons.length === 0) return;
+
+    const spacing = 28;
+    const baseY = -this.height / 2 - 5;
+
+    const totalWidth = (icons.length - 1) * spacing;
+    icons.forEach((icon, idx) => {
+      icon.x = -totalWidth / 2 + idx * spacing;
+      icon.y = baseY;
+    });
+  }
+
+  showDeviceIconFor(
+    iconKey: string,
+    emoji: string,
+    tooltipText: string | undefined,
+    durationMs = 2000,
+  ) {
+    this.showDeviceIcon(iconKey, emoji, tooltipText);
+    setTimeout(() => {
+      this.hideDeviceIcon(iconKey);
+    }, durationMs);
```

Comment:
We should make this take simulation speed into account. That way pausing the simulation makes the icons stay on screen.

```suggestion
    let progress = 0;
    const tick = (ticker: Ticker) => {
      progress += ticker.elapsedMS * this.ctx.getCurrentSpeed();
      if (progress >= durationMs) {
        Ticker.shared.remove(tick, this);
        this.hideDeviceIcon(iconKey);
      }
    };
    Ticker.shared.add(tick, this);
```